src/converse-chatview.js

```diff
@@ -344,11 +344,19 @@
                                "Output has been shortened."),
                             true, true);
                     }
+
+                    var mess_time;
+
+                    if(_converse.time_format === 'h')
+                        mess_time = msg_time.format('hh:mm a');
+                    else if(_converse.time_format === 'H')
+                        mess_time = msg_time.format('HH:MM');
+
                     var $msg = $(template(
                         _.extend(this.getExtraMessageTemplateAttributes(attrs), {
                             'msgid': attrs.msgid,
                             'sender': attrs.sender,
-                            'time': msg_time.format('hh:mm'),
+                            'time': mess_time,
```

Comment:
My original point had less to do with what the default was and more to do with the fact that you hardcoded the change.

I think I prefert 24-hour time as well, so I'd prefer HH:MM.

Concerning showing AM and PM, since we want to make the user (really the integrator) responsible for choosing the format, they can decide whether to show AM/PM or not.